mne/io/fiff/tests/test_raw.py

```diff
@@ -115,6 +115,34 @@ def test_copy_append():
     assert_equal(data.shape[1], 2 * raw._data.shape[1])
 
 
+def test_append_channels():
+    """Test raw splitting / re-appending channel types
+    """
+    raw = Raw(test_fif_fname, preload=True)
+    raw_nopre = Raw(fif_fname, preload=False)
+    raw_eeg_meg = raw.pick_types(meg=True, eeg=True, copy=True)
+    raw_eeg = raw.pick_types(meg=False, eeg=True, copy=True)
+    raw_meg = raw.pick_types(meg=True, eeg=False, copy=True)
+    raw_stim = raw.pick_types(meg=False, eeg=False, stim=True, copy=True)
+    raw_new = raw_meg.append_channels([raw_eeg])
+
+    assert_true([ch in raw_new.ch_names for ch in raw.ch_names])
+    assert_array_equal(raw_new[:, :][0], raw_eeg_meg[:, :][0])
+    assert_array_equal(raw_new[:, :][1], raw[:, :][1])
+    assert_true(all([ch not in raw_new.ch_names for ch in raw_stim.ch_names]))
```

Comment:
FYI you do not need the `[]` here. There is actually a difference -- with the `[]`, the whole list is generated, then `all` evaluated, whereas without it, `all` does the iterating, and therefore will short-circuit on the first non-`True` entry if it can. Thus it is more efficient not to have it. Here it doesn't matter because the computation is so simple with so few entries, but other times it can matter for efficiency.